### PR TITLE
feat(task-queue): add watchdog service, lease tracking, and Azure mod…

### PR DESCRIPTION
**Pull request by markus-lassfolk**
…el docs

## Summary

Closes #

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal improvement (no behavior change)
- [ ] Documentation update
- [ ] CI / tooling change

## Quality Checklist

Run from repo root (package lives under `extensions/memory-hybrid/`):

- [ ] `cd extensions/memory-hybrid && npx tsc --noEmit` passes with no errors
- [ ] `cd extensions/memory-hybrid && npm run lint` passes with no warnings
- [ ] `cd extensions/memory-hybrid && npm test` passes (all tests green)
- [ ] No secrets, credentials, or API keys committed
- [ ] No `console.log` debug statements left in production code

## Documentation Impact (REQUIRED)

- [ ] Inline code comments (JSDoc) updated for modified functions and types
- [ ] `docs/` or `README.md` updated to reflect architectural or usage changes
- [ ] Cross-referenced functions/services checked for outdated documentation

## Testing

- [ ] Unit tests added / updated
- [ ] Integration tests added / updated
- [ ] Manually verified against a running OpenClaw instance

## Notes for Reviewer

---

> [!NOTE]
> **Medium Risk**
> Adds a new durable dispatch-lease mechanism that changes how autonomous task-queue runs are deduped/expired and integrates it into the watchdog, which could affect queue progression under edge cases (locks/TTL/state transitions). Other changes are mostly safer defaults and doc/test updates around embeddings and provider validation.
> 
> **Overview**
> Introduces **durable task-queue dispatch leases** (`dispatch-leases.json` + lock file) to provide cross-process idempotency for autonomous issue dispatch, with TTL-based expiry, state transitions (`leased`→`running`→terminal), and an event trail.
> 
> Updates the **task-queue watchdog** to expire leases proactively and to force matching leases to a terminal state when it clears stale `current.json`, decoupling dedupe from GitHub branch/worktree propagation.
> 
> Adjusts verification/LLM plumbing: Google embedding fallback now uses `GOOGLE_EMBED_DEFAULT_MODEL` (`text-embedding-005`), `verify --test-llm` skips `azure-foundry` `gpt-5.4-pro` chat-completions tests, and `chatComplete` loosens typing to handle nonstandard response fields.
> 
> Adds/refreshes docs and examples (new `ARCHITECTURE-CENTER`, OpenClaw `models` schema workaround, troubleshooting config placement, Azure Foundry notes) and updates tests to cover leases and to skip dashboard bind tests when loopback binding is blocked.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b8b03f1baee729b78722aaf8b94290d0e097c93e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: extensions/memory-hybrid/tests/passive-observer.test.ts [hunk 1]
```diff
@@ -1129,16 +1129,16 @@ describe("isIdentityFact", () => {
...
   it("detects agent name without apostrophe when agentName is provided", () => {
-    expect(isIdentityFact("Doris email is doris@gmail.com", "Doris")).toBe(true);
+    expect(isIdentityFact("TestBot email is agent@example.com", "TestBot")).toBe(true);
   });
 
   it("is case-insensitive", () => {
-    expect(isIdentityFact("YOUR EMAIL IS DORIS@EXAMPLE.COM")).toBe(true);
-    expect(isIdentityFact("the ASSISTANT's name is doris", "doris")).toBe(true);
+    expect(isIdentityFact("YOUR EMAIL IS TESTBOT@EXAMPLE.COM")).toBe(true);
+    expect(isIdentityFact("the ASSISTANT's name is doris", "testbot")).toBe(true);
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/tests/passive-observer.test.ts]: Already fixed in previous commits (the agentName was changed from doris to testbot in a783cf8).
